app/decorators/models/solidus_easypost/spree/order_decorator.rb

```diff
@@ -3,29 +3,24 @@
 module Spree
   module OrderDecorator
     def self.prepended(base)
+      base.state_machine.before_transition to: :complete, do: :set_order_type
```

Comment:
You wrote: 
> Updated the state machine:
> The state machine now calls the set_order_type method after transitioning to the complete state.

I think we do this before complete or am I wrong?
The method should be called before the order is completed.